### ISSUE TITLE
Rewrite kfp-profile-controller to use Decorator controller

### ISSUE THREAD

**Issue by misohu** [canonical/kfp-operators#334]
### Bug Description

Right now kfp-profile-controller uses `CompositeController` as it is written for `metacontroller-operator` image version `v2.0.4`. This causes problem to `resource-dispatcher` as it uses `DecoratroController` written for `metacontroller-operator` image version `v3.0.0+` ([issue](https://github.com/canonical/resource-dispatcher/issues/30)). 

We should rewrite kfp-profile-controller to use  `DecoratroController` so we can update `metacontroller-operator` image version to `v3.0.0+`. More about why do we have to change to `DecoratroController` is described in this [closed issue](https://github.com/canonical/metacontroller-operator/issues/84).

In order to use `DecoratroController` in kfp-profile-controller we need to rewrite `files/upstream/sync.py` file. We can take inspiration from [resource-dispatcher-image source code](https://github.com/canonical/resource-dispatcher-image/tree/main/resource_dispatcher). 

### To Reproduce

None

### Environment

None

### Relevant log output

```shell
None
```

**Comment by misohu** [comment]
Fixed in https://github.com/canonical/kfp-operators/pull/342. 

**Ideas for improvements**
Currently we use sync.py file to specify the webhook's API. we push this file to the charm which makes it problematic:
- we have everything in one file
- we cannot have separate dependencies for the sync.py itself as it uses charm's one all the time
- we are not testing the API with unit tests we just relly on integration tests for charm 

The profile-controller itself has the same structure as  [resource-dispatcher ](https://github.com/canonical/resource-dispatcher). In there we decided to create a [separate image](https://github.com/canonical/resource-dispatcher-image) for the resource dispatcher api, instead of pushing file to the charm. This allow us to resolve all the problems above. We can use the same approach fot the profiles controller. We need to keep in mind that upstream is organizing code in [single sync.py file](https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/base/installs/multi-user/pipelines-profile-controller/sync.py) tho.
